File: LeetCodeTasks/45_jump-game-ii/solutions/c/solution.c
#include <limits.h>

int jump(int* nums, int numsSize) {
    if (numsSize <= 1) return 0;
    
    int jumps = 0, curEnd = 0, curFarthest = 0;
    for (int i = 0; i < numsSize - 1; i++) {
        curFarthest = (i + nums[i] > curFarthest) ? i + nums[i] : curFarthest;
        
        if (i == curEnd) {
            jumps++;
            curEnd = curFarthest;
            
            if (curEnd >= numsSize - 1) {
                break;
            }
        }
    }
    
    return jumps;
}

File: LeetCodeTasks/45_jump-game-ii/solutions/python3/solution.py
from typing import List

class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        if n <= 1:
            return 0
        
        jumps = 0
        current_end = 0
        farthest = 0
        for i in range(n):
            farthest = max(farthest, i + nums[i])
            if i == current_end:
                if i != n - 1:
                    jumps += 1
                    current_end = farthest
                    if current_end >= n - 1:
                        break
        return jumps
